src/engine/QueryPlanner.cpp

```diff
@@ -434,6 +407,96 @@ QueryExecutionTree QueryPlanner::createExecutionTree(ParsedQuery& pq) const {
   return *final._qet.get();
 }
 
+bool QueryPlanner::checkUsePatternTrick(
+    ParsedQuery& pq, SparqlTriple& patternTrickTriple) const {
+  bool usePatternTrick = false;
+  // Check if the query has the right number of variables for aliases and group
+  // by.
+  if (pq._groupByVariables.size() == 1 && pq._aliases.size() == 1) {
+    const ParsedQuery::Alias& alias = pq._aliases.back();
+    // Check if the alias is a non distinct count alias
+    if (alias._isAggregate &&
+        alias._function.find("DISTINCT") == std::string::npos &&
+        alias._function.find("distinct") == std::string::npos &&
+        (ad_utility::startsWith(alias._function, "COUNT") ||
+         ad_utility::startsWith(alias._function, "count"))) {
```

Comment:
can we do these checks on the lowercase `aliass._function` so it's actually case insensitive?